rotkehlchen/chain/evm/decoding/pendle/decoder.py

```diff
@@ -1,122 +1,317 @@
+            if event.amount == deposited_amount and event.event_type == HistoryEventType.SPEND:
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_FOR_WRAPPED
+                event.notes = f'Deposit {event.amount} {deposited_token.symbol} in a Pendle pool'
+
+            elif event.amount == received_amount and event.event_type == HistoryEventType.RECEIVE:
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Receive {received_amount} PENDLE-LPT for depositing in a Pendle pool'  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_remove_liquidity_event(self, context: DecoderContext) -> DecodingOutput:
+        received_amount = asset_normalized_value(
+            amount=int.from_bytes(context.tx_log.data[64:96]),
+            asset=(received_token := self.base.get_or_create_evm_token(bytes_to_address(context.tx_log.topics[3]))),  # noqa: E501
+        )
+        returned_amount = token_normalized_value_decimals(
+            token_amount=int.from_bytes(context.tx_log.data[32:64]),
+            token_decimals=DEFAULT_TOKEN_DECIMALS,
+        )
+        for event in context.decoded_events:
+            if event.amount == returned_amount and event.event_type == HistoryEventType.SPEND:
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.SPEND
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.notes = f'Return {event.amount} {event.asset.resolve_to_asset_with_symbol().symbol} to Pendle'  # noqa: E501
+
+            elif event.amount == received_amount and event.event_type == HistoryEventType.RECEIVE:
+                event.counterparty = CPT_PENDLE
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REDEEM_WRAPPED
+                event.notes = f'Withdraw {event.amount} {received_token.symbol} from a Pendle pool'
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_pt_yt_events(self, context: DecoderContext) -> DecodingOutput:
+        """Decode Pendle market events for buying or selling Principal (PT) or Yield (YT) tokens.
+
+        This method handles two primary scenarios:
+        - Buying PT/YT: token_0 is spent, token_1 is received
+        - Selling PT/YT: token_0 is received, token_1 is spent
+        """
+        market = bytes_to_address(context.tx_log.topics[2])
+        _, principal_token, yield_token = self.evm_inquirer.call_contract(
+            contract_address=market,
+            abi=PENDLE_ROUTER_ABI,
+            method_name='readTokens',
+        )
+        token_0 = (
+            A_ETH.resolve_to_crypto_asset()
```

Comment:
in binance do you also use A_ETH?

Reply:
switched to the `self.evm_inquirer.native_token`. thanks